lib/src/components/form/FormHelper.js

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { FormContext, statuses } from './Form';
+
+export function FormHelper({ children, className = '', ...rest }) {
+  const { status } = React.useContext(FormContext);
+  const hasFailed = status === statuses.failure;
+  const hasSucceeded = status === statuses.success;
+  const hasFailedRef = React.useRef(false);
+
+  if (hasFailed) {
+    hasFailedRef.current = true;
+  }
+
+  if (hasSucceeded) {
+    hasFailedRef.current = false;
+  }
+
+  return React.useMemo(
```

Comment:
🍏 

Would it be nicer to store `hasFailed` in the state, and have a `useEffect` triggering off a change in `status`. Which would then update the state? I suppose the actual question is, why does this have to be a ref?